core/trino-main/src/main/java/io/trino/sql/planner/QueryPlanner.java

```diff
@@ -793,21 +793,20 @@ public MergeWriterNode plan(Merge merge)
             List<ColumnHandle> mergeCaseSetColumns = mergeCaseColumnsHandles.get(caseNumber);
             for (ColumnHandle dataColumnHandle : mergeAnalysis.getDataColumnHandles()) {
                 int index = mergeCaseSetColumns.indexOf(dataColumnHandle);
+                int fieldNumber = requireNonNull(mergeAnalysis.getColumnHandleFieldNumbers().get(dataColumnHandle), "Field number for dataColumnHandle is null");
```

Comment:
```suggestion
int fieldNumber = requireNonNull(mergeAnalysis.getColumnHandleFieldNumbers().get(dataColumnHandle), "Could not find field number for column handle");
```

It is in sync with the message used in other places in this class.